### ISSUE TITLE
Proper location for Install for SSDT 2015

### ISSUE THREAD

**Issue by ACMjmorton** [keif888/SSISMHash#7]
I have done a Custom install of Multiple Hash, but it does not show up in the SSIS Tool box for Data Flows.  Please help me resolve this issue.  The specifics of my install and the actions I have taken are described below.

I installed SQL Server 2016 on a Windows 2012R2 64 bit server the same way I have for prior versions of SQL Server, where the Shared Feature directories are in C:\Program Files\Microsoft SQL Server and C:\Program Files (x86)\Microsoft SQL Server, but the Instance Root Directory is in D:\Program Files\MySQLinstanceName.

Then I installed the MultipleHashInstallerx64_20170713_V1.7.1.0.msi, also the same way I have for prior versions of Multiple Hash and SQL Server, choosing Custom Install and choosing C:\Program Files\Microsoft SQL Server\130\DTS  and C:\Program Files (x86)\Microsoft SQL Server\130\DTS for the Install locations.

After the install, I can see the MultipleHash2016.dll file in the PipelineComponents folder in the install locations.

When I open up SSDT 2015, I do not see Multiple Hash in the SSIS Toolbox Data Flow options.

Where should I have installed the Multiple Hash items?
Will I need to uninstall Multiple Hash and re-install it, or can I just move or copy the .dll files to the appropriate location?

If I installed them in the appropriate locations, what do I need to do so that Multiple Hash appears in the SSIS Toolbox for Data Flows?

I tried also installing MultipleHashInstallerx32_20170713_V1.7.1.0.msi, and I tried rebooting after the installs, but neither of those actions fixed the problem.

I also tried activating the Multiple Hash component within SSDT, but could not find a way to do this.
Right-clicking on "Common" or on "Other Transforms" (there is no "Data Flow Transformations" in SSDT 2015) gives only the option to "Refresh Toolbox".  I tried this, but Multiple Hash does not appear in the Toolbox.  I tried to activate it in SSDT using the Tools > Choose Toolbox Items menu options, but there is no "SSIS Data Flow Items" tab, and I did not find Multiple Hash on any of the tabs that did display.

**Comment by keif888** [comment]
Sorry for the delay in responding, I've been computer less for a couple of weeks.

I suspect that this is related to the issue with SSDT 17.2, which I am investigating at the moment.
I'll replicate your setup mentioned above, and check what happens.

The folder names you mention above look correct.

**Comment by keif888** [comment]
I have just completed virtual machine build, installation of SQL 2016 SP1 as noted above, and SSDT 17.2 (VS2015 version).
I then installed Multiple Hash 1.7.1, freshly downloaded from Github.
I opened SSDT.
I then created a new project, added a Data Flow, and Multiple Hash was NOT shown (I believe this replicated your scenario above).  This is expected with 1.7.1 as it does NOT support SQL 2017.
I then closed VS2015.
I opened SSDT.
I then created a new project, set the target version to SQL 2016, opened the default package, added a data flow, and Multiple Hash was shown.

Can you please confirm that the TargetServerVersion in the Configuration Properties for your SSIS project is set to SQL 2012, SQL 2014 or SQL 2016?

**Comment by keif888** [comment]
Release 1.7.2.0 has been made.  I believe that this addresses the issue encountered.  Please raise another issue if this is not the case.

**Comment by ACMjmorton** [comment]
Hi,
I finally got the chance to try changing the TargetServerVersion in the Configuration Properties for my SSIS Project to SQL 2016.  That, indeed, then included Multiple Hash in the Data Flow choices.

I then uninstalled v1.7.1.0 and installed v1.7.2.0 and Multiple Hash displays in the Data Flow choices without having to change the SSIS package TargetServerVersion.

Thank you for addressing this.
